tendenci/apps/memberships/views.py

```diff
@@ -2350,7 +2350,32 @@ def report_member_roster(request, template_name='reports/membership_roster.html'
 def report_member_quick_list(request, template_name='reports/membership_quick_list.html'):
     """ Table view of current members fname, lname and company only.
     """
-    members = MembershipDefault.objects.filter(status=1, status_detail="active").order_by('user__last_name')
+    order_by = request.GET.get('order_by', 'user__last_name')
+    
+    descending = order_by.startswith('-')
+    if descending:
+        order_by =  order_by[1:]
+    
+    members = MembershipDefault.objects.filter(status=1, status_detail='active')
+    
+    if order_by == 'member_number_int':
+        members = members.annotate(member_number_int=Cast('member_number', PositiveIntegerField()))
+
+    # If we want to be able order by an aggregated list of groups
+    # then we need to annotate the with that. Django provides no 
+    # generic means for creating such aggregates yet but the postgresql
+    # package provides one. So we can order by the group list only on
+    # postgresql systems for now. 
+    #
+    # Note the template can render a list without this annotation and
+    # shoudl fall on such a method.  
+    if connection.vendor == 'postgresql':
+        members = members.annotate(user_group_list=StringAgg('user__user_groups__name', ', ', ordering='user__user_groups__id'))
+    
+    if descending:
+        order_by = '-' + order_by
+        
+    members = members.order_by(order_by)
```

Comment:
The `order_by` needs to be validated to ensure it contains the intended value before it can be used in the query.

Reply:
Oops, forgot to stage and commit the tag. Fixed.

Reply:
Have you any tips on how? Am struggling to find a validator for field lookups at present. Basically fields can be on membership or membership.user or even deeper in theory, but there's a syntax for field lookups such that `user__lastname` is a reference to a field on `membership.user`. I'd like to think a general purpous validator for such field lookups already exists, but I can't find one yet.